contracts/FlowTransactionSchedulerUtils.cdc

```diff
@@ -0,0 +1,335 @@
+                } else {
+                    self.idsByTimestamp[timestamp] = ids
+                }
+            }
+
+            if let ids = self.idsByHandler[handlerTypeIdentifier] {
+                let index = ids.firstIndex(of: id)
+                ids.remove(at: index!)
+                self.idsByHandler[handlerTypeIdentifier] = ids
+            }
+        }
+
+        /// Clean up transactions that are no longer valid (return nil or Unknown status)
+        /// This removes and destroys transactions that have been executed, canceled, or are otherwise invalid
+        /// @return: The transactions that were cleaned up (removed from the manager)
+        access(Owner) fun cleanup(): [UInt64] {
+            let currentTimestamp = getCurrentBlock().timestamp
+            var transactionsToRemove: [UInt64] = []
+
+            let pastTimestamps = self.sortedTimestamps.getBefore(current: currentTimestamp)
+            for timestamp in pastTimestamps {
+                let ids = self.idsByTimestamp[timestamp] ?? []
+                for id in ids {
+                    let status = FlowTransactionScheduler.getStatus(id: id)
+                    if status == nil || status == FlowTransactionScheduler.Status.Unknown {
+                        transactionsToRemove.append(id)
+                    }
+                }
+            }
+
+            // Then remove and destroy the identified transactions
+            for id in transactionsToRemove {
+                if let tx <- self.scheduledTransactions.remove(key: id) {
+                    self.removeID(id: id, timestamp: tx.timestamp, handlerTypeIdentifier: tx.handlerTypeIdentifier)
+                    destroy tx
+                }
+            }
+
+            return transactionsToRemove
+        }
+
+        /// Get transaction data by its ID
+        /// @param id: The ID of the transaction to retrieve
+        /// @return: The transaction data from FlowTransactionScheduler, or nil if not found
+        access(all) view fun getTransactionData(id: UInt64): FlowTransactionScheduler.TransactionData? {
+            if self.scheduledTransactions.containsKey(id) {
+                return FlowTransactionScheduler.getTransactionData(id: id)
+            }
+            return nil
+        }
+
+        /// Get an un-entitled reference to a transaction handler of a given ID
+        /// @param id: The ID of the transaction to retrieve
+        /// @return: A reference to the transaction handler, or nil if not found
+        access(all) view fun getTransactionHandler(id: UInt64): &{FlowTransactionScheduler.TransactionHandler}? {
+            let txData = self.getTransactionData(id: id)
+            return txData?.getUnentitledHandlerReference()
+        }
+
+        /// Get all the handler type identifiers that the manager has transactions scheduled for
+        /// @return: An array of all handler type identifiers
+        access(all) view fun getHandlerTypeIdentifiers(): [String] {
```

Comment:
Would it be possible to have a method to retrieve handler unentitled reference by type identifier?
This would allow us to show all the handlers data from an address:

1 - we get handler type identifiers by address (getHandlerTypeIdentifiers())
2 - then get handler reference from which we can access views for each handler (getHandlerByTypeIdentifier(typeIdentifier))

Reply:
Yeah, we can add that for the manager, but not globally unless you know the address of which manager you're querying